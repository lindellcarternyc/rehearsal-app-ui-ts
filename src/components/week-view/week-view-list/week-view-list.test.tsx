import * as React from 'react'
import * as renderer from 'react-test-renderer'

import WeekViewList, { WeekViewListProps } from './week-view-list'

describe('WeekViewList', () => {
  it('renders a week with no rehearsals', () => {
    const props: WeekViewListProps = {
      days: [
        {
          date: 'Sun, January 7'
        },
        {
          date: 'Mon, January 8'
        },
        {
          date: 'Tue, January 9'
        },
        {
          date: 'Wed, January 10'
        },
        {
          date: 'Thu, January 11'
        },
        {
          date: 'Fri, January 12'
        },
        {
          date: 'Sat, January 13'
        }
      ]
    }

    const tree = renderer.create(
      <WeekViewList {...props} />
    ).toJSON()
    expect(tree).toMatchSnapshot()
  })

  it('renders a week with rehearsals', () => {
    const props: WeekViewListProps = {
      days: [
        {
          date: 'Sun, January 14'
        },
        {
          date: 'Mon, January 15'
        },
        {
          date: 'Tue, January 16',
          rehearsals: [
            {time: '12:00 - 14:00', material: ['']}
          ]
        },
        {
          date: 'Wed, January 17'
        },
        {
          date: 'Thu, January 18'
        },
        {
          date: 'Fri, January 19'
        },
        {
          date: 'Sat, January 20',
          rehearsals: [
            {time: '13:00 - 14:30', material: ['']},
            {time: '14:30 - 16:00', material: ['']},
            {time: '16:00 - 18:00', material: ['']}
          ]
        }
      ]
    }
    const tree = renderer.create(
      <WeekViewList {...props} />
    ).toJSON()
    expect(tree).toMatchSnapshot()
  })
})