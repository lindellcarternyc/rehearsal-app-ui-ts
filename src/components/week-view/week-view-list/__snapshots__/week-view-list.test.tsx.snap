// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`WeekViewList renders a week with no rehearsals 1`] = `
<div
  className="ui celled relaxed list"
  role="list"
>
  <div
    className="item"
    data-value={undefined}
    onClick={[Function]}
    role="listitem"
  >
    <div
      className="header"
    >
      Sun, January 7
    </div>
    No Rehearsals
  </div>
  <div
    className="item"
    data-value={undefined}
    onClick={[Function]}
    role="listitem"
  >
    <div
      className="header"
    >
      Mon, January 8
    </div>
    No Rehearsals
  </div>
  <div
    className="item"
    data-value={undefined}
    onClick={[Function]}
    role="listitem"
  >
    <div
      className="header"
    >
      Tue, January 9
    </div>
    No Rehearsals
  </div>
  <div
    className="item"
    data-value={undefined}
    onClick={[Function]}
    role="listitem"
  >
    <div
      className="header"
    >
      Wed, January 10
    </div>
    No Rehearsals
  </div>
  <div
    className="item"
    data-value={undefined}
    onClick={[Function]}
    role="listitem"
  >
    <div
      className="header"
    >
      Thu, January 11
    </div>
    No Rehearsals
  </div>
  <div
    className="item"
    data-value={undefined}
    onClick={[Function]}
    role="listitem"
  >
    <div
      className="header"
    >
      Fri, January 12
    </div>
    No Rehearsals
  </div>
  <div
    className="item"
    data-value={undefined}
    onClick={[Function]}
    role="listitem"
  >
    <div
      className="header"
    >
      Sat, January 13
    </div>
    No Rehearsals
  </div>
</div>
`;

exports[`WeekViewList renders a week with rehearsals 1`] = `
<div
  className="ui celled relaxed list"
  role="list"
>
  <div
    className="item"
    data-value={undefined}
    onClick={[Function]}
    role="listitem"
  >
    <div
      className="header"
    >
      Sun, January 14
    </div>
    No Rehearsals
  </div>
  <div
    className="item"
    data-value={undefined}
    onClick={[Function]}
    role="listitem"
  >
    <div
      className="header"
    >
      Mon, January 15
    </div>
    No Rehearsals
  </div>
  <div
    className="item"
    data-value={undefined}
    onClick={[Function]}
    role="listitem"
  >
    <div
      className="header"
    >
      Tue, January 16
    </div>
    No Rehearsals
  </div>
  <div
    className="item"
    data-value={undefined}
    onClick={[Function]}
    role="listitem"
  >
    <div
      className="header"
    >
      Wed, January 17
    </div>
    No Rehearsals
  </div>
  <div
    className="item"
    data-value={undefined}
    onClick={[Function]}
    role="listitem"
  >
    <div
      className="header"
    >
      Thu, January 18
    </div>
    No Rehearsals
  </div>
  <div
    className="item"
    data-value={undefined}
    onClick={[Function]}
    role="listitem"
  >
    <div
      className="header"
    >
      Fri, January 19
    </div>
    No Rehearsals
  </div>
  <div
    className="item"
    data-value={undefined}
    onClick={[Function]}
    role="listitem"
  >
    <div
      className="header"
    >
      Sat, January 20
    </div>
    No Rehearsals
  </div>
</div>
`;
